can/interfaces/gs_usb.py

```diff
@@ -0,0 +1,107 @@
+        if not gs_usb:
+            raise can.CanError("Can not find device {}".format(channel))
+        self.gs_usb = gs_usb
+        self.channel_info = channel
+
+        self.gs_usb.set_bitrate(bitrate)
+        self.gs_usb.start()
+
+        super().__init__(channel=channel, can_filters=can_filters, **kwargs)
+
+    def send(self, msg: can.Message, timeout: Optional[float] = None):
+        """Transmit a message to the CAN bus.
+
+        :param Message msg: A message object.
+        :param timeout: timeout is not used here
+
+        :raises can.CanError:
+            if the message could not be sent
+        """
+        can_id = msg.arbitration_id
+
+        if msg.is_extended_id:
+            can_id = can_id | CAN_EFF_FLAG
+
+        if msg.is_remote_frame:
+            can_id = can_id | CAN_RTR_FLAG
+
+        if msg.is_error_frame:
+            can_id = can_id | CAN_ERR_FLAG
+
+        frame = GsUsbFrame()
+        frame.can_id = can_id
+        frame.can_dlc = msg.dlc
+        frame.timestamp_us = int(msg.timestamp * 1000000)
+        msg.data.extend([0x00] * (CAN_MAX_DLC - len(msg.data)))
+        frame.data = list(msg.data)
+
+        self.gs_usb.send(frame)
+
+    def _recv_internal(
+        self, timeout: Optional[float]
+    ) -> Tuple[Optional[can.Message], bool]:
+        """
+        Read a message from the bus and tell whether it was filtered.
+        This methods may be called by :meth:`~can.BusABC.recv`
+        to read a message multiple times if the filters set by
+        :meth:`~can.BusABC.set_filters` do not match and the call has
+        not yet timed out.
+
+        :param float timeout: seconds to wait for a message,
+                              see :meth:`~can.BusABC.send`
+                              0 and None will be converted to minimum value 1ms.
+                              float will be finally converted to integer.
+
+        :return:
+            1.  a message that was read or None on timeout
+            2.  a bool that is True if message filtering has already
+                been done and else False. In this interface it is always False
+                since filtering is not available
+
+        :raises can.CanError:
+            if an error occurred while reading
+        """
+        frame = GsUsbFrame()
+
+        # Do not set timeout as None or zero here to avoid blocking
+        timeout_ms = round(timeout * 1000) if timeout else 1
+        if not self.gs_usb.read(frame=frame, timeout_ms=timeout_ms):
+            return None, False
+
+        msg = can.Message()
+        msg.arbitration_id = frame.arbitration_id
+        msg.dlc = frame.can_dlc
+        msg.data = bytearray(frame.data)[0 : msg.dlc]
+        msg.timestamp = frame.timestamp
+        msg.channel = self.channel_info
+        msg.is_extended_id = frame.is_extended_id
+        msg.is_remote_frame = frame.is_remote_frame
+        msg.is_error_frame = frame.is_error_frame
+        msg.is_rx = True
```

Comment:
Can you please pass these as named arguments to the `Message` constructor

Reply:
Done